dissect/cstruct/parser.py

```diff
@@ -294,16 +294,16 @@ def _names(self, tokens: TokenConsumer) -> List[str]:
     @staticmethod
     def _remove_comments(string: str) -> str:
         # https://stackoverflow.com/a/18381470
-        pattern = r"(\".*?\"|\'.*?\')|(/\*.*?\*/|//[^\r\n]*$)"
+        pattern = r"([\"\'].*?[\"\'])|(/\*.*?\*/|//[^\r\n]*$)"
```

Comment:
It now also triggers on this `"<anything other than ' or ">'` situation or vice versa, is that what you wanted to include here?

So to elaborate, if someone uses the comment: `"This is a 'tricky' comment"` the first group will be `"This is a '` since `.*` is not greedy.

Reply:
It's indeed different, reverted the pattern.